src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,83 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.Importance;
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.domain.User;
+import com.tstepnik.planner.exceptions.TaskNotFoundException;
+import com.tstepnik.planner.repository.TaskRepository;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.MINOR;
+
+    private final TaskRepository taskRepository;
+    private final AuthService authService;
+//TODO here add whole logic with if and exception and use loggedUser here!
+
+
+    public TaskService(TaskRepository taskRepository, AuthService authService) {
+        this.taskRepository = taskRepository;
+        this.authService = authService;
+    }
+
+    public List<Task> findAll() {
+        return taskRepository.findAll();
+    }
+
+    public Optional<Task> findTaskById(Long id) {
+        return taskRepository.findById(id);
+    }
+
+    public List<Task> getUserTasks() {
+        User user = authService.getLoggedUser();
+        return taskRepository.findAllByUserId(user.getId());
+    }
+
+    public Task addTask(Task task) {
+        User user = authService.getLoggedUser();
+        if (task.getImportance() == null) {
+            task.setImportance(DEFAULT_IMPORTANCE);
+        }
+        task.setUserId(user.getId());
+        return taskRepository.save(task);
+    }
+
+    public Task updateTask(Task task, Long taskId) {
+        User user = authService.getLoggedUser();
+        Optional<Task> checkedTask = findTaskById(taskId);
+        if (checkedTask.isEmpty()) {
+            return taskRepository.save(task);
+        } else if (checkedTask.get().getUserId().equals(user.getId())) {
+            Task updatedTask = checkedTask.get();
+            updatedTask.setImportance(task.getImportance());
+            updatedTask.setDescription(task.getDescription());
+            updatedTask.setDone(task.isDone());
+            return taskRepository.save(updatedTask);
+        }
+        throw new AccessDeniedException("This task doesn't belongs to User");
```

Comment:
This is not really clear when this exception will be thrown. Try to formulate code so throwing exception is done before happy path code. 
```
if(user is not owner)
	throw exception
else
	update
	return updatedTask
```

Reply:
I had done it previously but once we were talking about good pracites and you said that at first should be happy path and exception always at the end because it's easier for reading so i change it to actual solution, but I will change it ;)

Reply:
Did I really say this? I am not sure, because starting like 6 months ago I switched to the solution described in the comment above.

Reply:
Yes, but acctual approach is more readable i think.

Reply:
I think it can be more cleaner.